src/Model.js

```diff
@@ -34,29 +40,42 @@ export default class Model {
   }
 
   *save() {
-    this[exec_middlewares]('before', 'save')
-    // TODO: save model
-    this[exec_middlewares]('after', 'save')
+    this[execute_middleware]('before', 'save')
+
+    // Build collection instance
+    let obj = Object.assign({}, this)
+    obj._id = obj.id
+    delete obj.id
 
-    // TODO: replace {} with the query result
-    yield {}
+    yield* this[collection].update(obj._id, obj, { upsert: true })
+    this[execute_middleware]('after', 'save')
+
+    return this
   }
 
   *delete() {
-    this[exec_middlewares]('before', 'delete')
-    // TODO: delete model
-    this[exec_middlewares]('after', 'delete')
-
-    // TODO: replace {} with the query result
-    yield {}
+    this[execute_middleware]('before', 'delete')
+    yield* this[collection].removeById(this.id)
+    this[execute_middleware]('after', 'delete')
+    return this
   }
 
-  [exec_middlewares](timing, action) {
+  [execute_middleware](timing, action) {
     this[middlewares][timing][action].forEach(middleware => {
       middleware.call(this)
     })
   }
 
+  static get_collection() {
+    let database = monk('localhost/app')
```

Comment:
This path shouldn't be hardcoded here, the user must be able to choose which db to use